src/main/java/org/springframework/data/elasticsearch/repository/query/ElasticsearchStringQuery.java

```diff
@@ -103,4 +115,35 @@ protected StringQuery createQuery(ParametersParameterAccessor parameterAccessor)
 		return new StringQuery(queryString);
 	}
 
+	/**
+	 * Parse the {@link SourceFilters} attributes to construct a SourceFilter {@link SourceFilter}
+	 * @param parameterAccessor the accessor with the query method parameter details
+	 * @return source filter with includes and excludes for a  query
+	 * @since 4.4
+	 */
+	private SourceFilter processSourceFilterParams(SourceFilters sourceFilters, ParameterAccessor parameterAccessor) {
+		StringQueryUtil stringQueryUtil = new StringQueryUtil(elasticsearchOperations.getElasticsearchConverter().getConversionService());
+		ObjectMapper objectMapper = new ObjectMapper();
+		FetchSourceFilterBuilder fetchSourceFilterBuilder = new FetchSourceFilterBuilder();
+		String errorMessage = null;
+		String includesInput = stringQueryUtil.replacePlaceholders(sourceFilters.includes(), parameterAccessor);
+		String excludesInput = stringQueryUtil.replacePlaceholders(sourceFilters.excludes(), parameterAccessor);
+		try {
+			if (!includesInput.equals("")) {
+				String[] includes = objectMapper.readValue(includesInput, String[].class);
+				fetchSourceFilterBuilder.withIncludes(includes);
+			}
+			if (!excludesInput.equals("")) {
+				String[] excludes = objectMapper.readValue(excludesInput, String[].class);
+				fetchSourceFilterBuilder.withExcludes(excludes);
+			}
+		} catch (JsonProcessingException e) {
+			errorMessage = e.getMessage();
+		}
+
+		SourceFilter sourceFilter = fetchSourceFilterBuilder.build();
+		Assert.isTrue(sourceFilter.getIncludes().length > 0 || sourceFilter.getExcludes().length > 0,
+				"At least one includes or excludes should be provided.\n Found error: " + errorMessage);
+		return sourceFilter;
+	}
```

Comment:
I prefer the simplest solution, which in this case does not involve parameter interpolation at each element of the array. It is also in keeping with the `@Query` annotation of using string values instead of the underlying data structure (i.e. `@Query.value` is a string instead of a map)